pkg/decoder/decoder.go

```diff
@@ -52,25 +74,55 @@ func (d *Decoder) Start() {
 	go d.run()
 }
 
+// Stop stops the Decoder
+func (d *Decoder) Stop() {
+	close(d.InputChan)
+}
+
 // run lets the Decoder handle data coming from the InputChan
 func (d *Decoder) run() {
 	for data := range d.InputChan {
-		d.decodeIncomingData(data.content, data.offset)
+		d.stopCountDown()
+		if ok, offset := d.decodeIncomingData(data.content, data.offset); ok {
+			d.restartCountDown(offset)
+		}
 	}
 	d.OutputChan <- message.NewStopMessage()
 }
 
-// Stop stops the Decoder
-func (d *Decoder) Stop() {
-	close(d.InputChan)
+// defaultCountDownDuration represents the time after which the buffered line is sent when there is no more incoming data
+const defaultCountDownDuration = 1 * time.Second
+
+// stopCountDown prevents the timer for firing
+func (d *Decoder) stopCountDown() {
+	if d.singleline {
+		return
+	}
+	if d.timer != nil {
+		d.timer.Stop()
+	}
+}
+
+// restartCountDown starts the timer and sends the last line when it fires
+func (d *Decoder) restartCountDown(offset int64) {
+	if d.singleline {
+		return
+	}
+	d.timer = time.AfterFunc(d.countDownDuration, func() {
+		newLine := make([]byte, d.lineBuffer.Len())
```

Comment:
I would extract this function. Also could it be code duplication?

I'd do something like `time.AfterFunc(..., d.flushBuffer)`